frontend/src/data/react/WithLoadingOrErrorUI.tsx

```diff
@@ -120,7 +143,21 @@ export function WithMetadataAndMetricsWithKey(
           queries={props.queries}
           loadingComponent={props.loadingComponent}
         >
-          {(queryResponses) => props.children(metadata, queryResponses)}
+          {(queryResponses) => {
+            if (!props.loadGeographies) {
+              return props.children(metadata, queryResponses);
+            }
+            return (
+              <WithDatasets
+                datasetIds={[GEOGRAPHIES_DATASET_ID]}
+                loadingComponent={props.loadingComponent}
+              >
+                {(datasets) =>
+                  props.children(metadata, queryResponses, datasets[0].rows)
```

Comment:
Using datasets[0] feels a bit fragile - any other way to pull out the geographies dataset here?

Reply:
I think this is an existing problem with these utility components - they all take an array of things to load and return an array of values expected in the same order (modeled after Promise.all). I changed to array destructuring to make it more explicit, and added a TODO to consider changing to a map since I'd want to update all of these at the same time for consistency.